package/network/services/wireguard/files/wireguard.sh

```diff
@@ -147,13 +148,13 @@ proto_wireguard_setup() {
 				proto_add_ipv6_address "${address%%/*}" "${address##*/}"
 				;;
 			*.*/*)
-				proto_add_ipv4_address "${address%%/*}" "${address##*/}"
+				proto_add_ipv4_address "${address%%/*}" "${address##*/}" "" "${peeraddr}"
```

Comment:
Does `proto_add_ipv4_address` act differently when the last two arguments are `""`? That is to ask, does it use `$#` in any way that might change the behavior for users not using this parameter?

Reply:
No, proto_add_ipv4_address ignores empty parameters.

```
root@client:~# tail -15 /etc/config/network 
config interface 'wg0'
	option proto 'wireguard'
	option private_key 'XXXXXXXXXXX='
	option addresses '192.168.253.12'
	#option peeraddr '192.168.253.1'

config wireguard_wg0
	option public_key 'ZZZZZZZZZ='
	option endpoint_host 'somehost.com'
	option endpoint_port '51820'
	option persistent_keepalive '29'
	list allowed_ips '0.0.0.0/0'
	list allowed_ips '::/0'

root@client:~# ifup wg0
root@client:~# ip a show dev wg0
15: wg0: <POINTOPOINT,NOARP,UP,LOWER_UP> mtu 1420 qdisc noqueue state UNKNOWN group default qlen 1000
    link/none 
    inet 192.168.253.12/32 brd 255.255.255.255 scope global wg0
       valid_lft forever preferred_lft forever

root@client:~# ifdown wg0
root@client:~# nano /etc/config/network 
root@client:~# tail -15 /etc/config/network 

config interface 'wg0'
	option proto 'wireguard'
	option private_key 'XXXXXXXXXX='
	option addresses '192.168.253.12'
	option peeraddr '192.168.253.1'

config wireguard_wg0
	option public_key 'ZZZZZZZZZZ='
	option endpoint_host 'somehost.com'
	option endpoint_port '51820'
	option persistent_keepalive '29'
	list allowed_ips '0.0.0.0/0'
	list allowed_ips '::/0'

root@client:~# ifup wg0
root@client:~# ip a show dev wg0
16: wg0: <POINTOPOINT,NOARP,UP,LOWER_UP> mtu 1420 qdisc noqueue state UNKNOWN group default qlen 1000
    link/none 
    inet 192.168.253.12 peer 192.168.253.1/32 brd 255.255.255.255 scope global wg0
       valid_lft forever preferred_lft forever

```